pmc/journal_recommendation/journal_recommendation_figure_generator.py

```diff
@@ -0,0 +1,570 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Journal Recommendation - Figure Generator
+
+# In[1]:
+
+
+from collections import Counter
+import itertools
+import os
+from pathlib import Path
+import pickle
+import random
+
+import numpy as np
+import pandas as pd
+from scipy.spatial.distance import cdist
+from sklearn.decomposition import PCA
+from tqdm import tqdm
+
+from saucie_modules import SAUCIE, Loader
+import tensorflow.compat.v1 as tf
+
+import plotnine as p9
+
+
+# In[2]:
+
+
+# Set up porting from python to R
+# and R to python :mindblown:
+
+import rpy2.rinterface
+get_ipython().run_line_magic('load_ext', 'rpy2.ipython')
+
+
+# # Plot Accuracy Results
+
+# In[3]:
+
+
+results = [
+    {"value":.00517, "model":"random_baseline", "distance":"N/A", "dataset":"train"},
+    {"value":.39982, "model":"paper_paper", "distance":"euclidean", "dataset":"train"},
+    {"value":.37340, "model":"centroid", "distance":"euclidean", "dataset":"train"},
+    {"value":.39824, "model":"paper_paper", "distance":"manhattan", "dataset":"train"},
+    {"value":.39824, "model":"centroid", "distance":"manhattan", "dataset":"train"},
+    {"value":.00445, "model":"random_baseline", "distance":"N/A", "dataset":"test"},
+    {"value":.20638, "model":"paper_paper", "distance":"euclidean", "dataset":"test"},
+    {"value":.21636, "model":"centroid", "distance":"euclidean", "dataset":"test"},
+    {"value":.20380, "model":"paper_paper", "distance":"manhattan", "dataset":"test"},
+    {"value":.21523, "model":"centroid", "distance":"manhattan", "dataset":"test"}
+]
+
+
+# In[4]:
+
+
+result_df = pd.DataFrame.from_records(results)
+
+result_df['dataset'] = pd.Categorical(
+    result_df.dataset.tolist(), 
+    categories=['train', 'test']
+)
+
+result_df.head()
+
+
+# In[5]:
+
+
+g = (
+    p9.ggplot(result_df.query("distance in ['euclidean', 'N/A']"), p9.aes(x="model", y="value"))
```

Comment:
So there's not a difference between recommending using the dist(your paper, paper in journal X) vs dist(your paper, centroid of all papers in journal X)?

Reply:
Does that mean that papers within a journal are pretty robust/similar? I thought you mentioned these big journals with a lot of different topics..maybe these were removed?

Reply:
>So there's not a difference between recommending using the dist(your paper, paper in journal X) vs dist(your paper, centroid of all papers in journal X)?

Test set wise there isn't a strong difference between both classifiers.

>Does that mean that papers within a journal are pretty robust/similar? I thought you mentioned these big journals with a lot of different topics..maybe these were removed?

No I don't think this result supports that claim. The reason for decent performance is that the mega-journals are being classified correctly for most papers that originated from them. If you take the mega-journals out performance drops.